packages/network-controller/src/NetworkController.test.ts

```diff
@@ -1,224 +1,3653 @@
-import * as sinon from 'sinon';
-import Web3ProviderEngine from 'web3-provider-engine';
+import assert from 'assert';
+import { mocked } from 'ts-jest/utils';
 import { ControllerMessenger } from '@metamask/base-controller';
-import { NetworkType, NetworksChainId } from '@metamask/controller-utils';
+import * as ethQueryModule from 'eth-query';
+import Subprovider from 'web3-provider-engine/subproviders/provider';
+import createInfuraProvider from 'eth-json-rpc-infura/src/createProvider';
+import type { ProviderEngine } from 'web3-provider-engine';
+import createMetamaskProvider from 'web3-provider-engine/zero';
+import { Patch } from 'immer';
+import {
+  FakeProviderEngine,
+  FakeProviderStub,
+} from '../tests/fake-provider-engine';
 import {
   NetworkController,
-  NetworkControllerMessenger,
+  NetworkControllerActions,
+  NetworkControllerEvents,
   NetworkControllerOptions,
+  NetworkState,
   ProviderConfig,
 } from './NetworkController';
 
-const RPC_TARGET = 'http://foo';
-
-const setupController = (
-  pType: NetworkType,
-  messenger: NetworkControllerMessenger,
-) => {
-  const networkControllerOpts: NetworkControllerOptions = {
-    infuraProjectId: 'foo',
-    state: {
-      network: '0',
-      providerConfig: {
-        type: pType,
-        chainId: NetworksChainId[pType],
-      },
-      properties: { isEIP1559Compatible: false },
-    },
-    messenger,
+jest.mock('eth-query', () => {
+  return {
+    __esModule: true,
+    default: jest.requireActual('eth-query'),
   };
-  const controller = new NetworkController(networkControllerOpts);
-  controller.providerConfig = {} as ProviderConfig;
-  return controller;
-};
+});
+jest.mock('web3-provider-engine/subproviders/provider');
+jest.mock('eth-json-rpc-infura/src/createProvider');
+jest.mock('web3-provider-engine/zero');
 
-describe('NetworkController', () => {
-  let messenger: NetworkControllerMessenger;
+type WithControllerCallback<ReturnValue> = ({
+  controller,
+}: {
+  controller: NetworkController;
+}) => Promise<ReturnValue> | ReturnValue;
+
+type WithControllerOptions = Partial<NetworkControllerOptions>;
+
+type WithControllerArgs<ReturnValue> =
+  | [WithControllerCallback<ReturnValue>]
+  | [WithControllerOptions, WithControllerCallback<ReturnValue>];
 
+// Store this in case it gets stubbed later
+const originalSetTimeout = global.setTimeout;
```

Comment:
Nit: A newline here might help readability, so it's clear this comment only applies to this one line

```suggestion
const originalSetTimeout = global.setTimeout;

```

Reply:
Fixed in [58964c5](https://github.com/MetaMask/core/pull/1039/commits/58964c55b78ceda9061dfb316dfcdb7e36bb2a5f).